components/script/dom/performance.rs

```diff
@@ -113,6 +114,25 @@ impl PerformanceEntryList {
             None => 0.,
         }
     }
+    //clears the buffer
+    fn clearResourceTimings(&mut self){
+         self.entries.clear();
```

Comment:
This would clear all the registered entries. We don't want that. What we want is to clear only the ones with `PerformanceResourceTiming` type.

We have the `clear_entries_by_name_and_type` to help with this task. So you could do something like:

```rust
self.clear_entries_by_name_and_type(None, Some(DOMString::from("resource")));
```

This method belongs to the `Performance` interface, and so it should be implemented in the `Performance` struct.